### PR TITLE
bazel: add utility macro for wrapping single-file tools

### PR DESCRIPTION
**Pull request by Strum355**
Currently, we provide single-file tools such as `ctags`, `gsutil` etc via an `sh_binary` wrapper, to have a single target to reference that automatically does platform selection of the underlying tool. 
Due to some [unfortunate reason](https://github.com/bazelbuild/bazel/issues/11820), the underlying srcs (which is [a single file](https://bazel.build/reference/be/shell#sh_binary.srcs)) of an `sh_binary` are also exposed as outputs (rather than just as typical runfiles) alongside the script that wraps. This is _sometimes_ problematic when doing location expansion (e.g. `$(location ...)`) due to these only allowing a single output (dont ask why this works in some contexts but not others, I dont know). 
To address this, we create a wrapper macro + rule to replicate what we want from `sh_binary` (automatic platform selection + tool naming), while only exposing a singular file.

See example of currently required approach to consuming a tool: [BUILD.bazel](https://github.com/sourcegraph/sourcegraph/pull/62801/files#diff-e2a562c2e13908933b2ee24f0ac596829b38a5325cc69a4aee05c383aaa2e494R8) & [main_test.go](https://github.com/sourcegraph/sourcegraph/pull/62801/files#diff-7a91cb5143064bfc8993ef97baf68b718ef49747ccc1d3c5e1150d4696b88305R66).

With this change, `rlocationpath` (singular) can be used instead (or any of the other singular nouns in different contexts), as well as no `strings.Split/strings.Fields` being required

## Test plan

`bazel cquery --output=files //dev/tools:dropdb` yields 1 vs 2 files.
Also updated the rule behind `//internal/database:generate_schemas` due to the workaround in it for the fact that the underlying srcs was also exposed. The correctness is verified by running said target (locally + CI)

### CODE REVIEW

File: dev/tools/tool.bzl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+# we want to rename. For example, files produced by http_file have the name `downloaded`,
+# we may want to rename them with sh_binary, however this has a weird quirk whereby both
+# the sh_binary wrapper "script" and the input binary/script itself are included in the outputs.
+# This results in some less ergonomic usage when trying to use them in e.g. go tests, having to
+# use $(rlocationpaths ...) and then filepath.Dir(runfiles.Rlocation(strings.Split(..., " ")[0])) in
+# order to get the path.
+# With this macro, its slightly simplified by being able to use $(rlocationpath ...) (singular) and
+# not having to strings.Split(..., " ") as a result.
+# See more: https://github.com/bazelbuild/bazel/issues/11820
+def tool(name, src, visibility = ["//visibility:public"]):
```
Comment (jhchabran) [inline dev/tools/tool.bzl:13]: Aren't we breaking conventions here? I'm not exactly sure what depends on it, but in various places in the docs, it says that things should be named `_test` or `_binary` for example.

Comment (Strum355) [reply dev/tools/tool.bzl:13]: That requirement exists for test _rules_ only (see docs here: https://bazel.build/extending/rules#executable_rules_and_test_rules). There is loosely [this](https://bazel.build/reference/be/common-definitions#common-attributes-binaries) for binary targets, but even this isnt widely honoured (go_binary doesnt support `args` for example)

**Comment by jhchabran** [comment]
@Strum355 in case you didn't see it, there's a breakage on pgutils.

File: dev/migrations.bzl [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # Because we have a bunch of ${} in there, which clash with the interpolation for "".format(...),
 # it's simpler to have this in a var and inject it as is, rather than having to escape everything.
-CMD_PREAMBLE = """set -e
+CMD_PREAMBLE = """set -ex
```
Comment (jhchabran) [inline dev/migrations.bzl:3]: Do we want to keep this? Or was that to debug 😊 ?

**Review by jhchabran — APPROVED** [review]

File: dev/migrations.bzl (hunk 2 above)
Comment (Strum355) [reply dev/migrations.bzl:3]: I keep forgetting this when I add it for debugging 🙃

Comment (Strum355) [reply dev/migrations.bzl:3]:
```suggestion
CMD_PREAMBLE = """set -e
```

**Pull request merged**